### DISCUSSION TITLE
Support for Seblod CCK

### DISCUSSION THREAD

**Discussion by raramuridesign** [joomlatools/joomlatools-pages#713] [General]
Would it be possible to support SEBLOD CCK? See: https://www.seblod.com/
M

**Comment by raramuridesign** [comment]
Thanks @johanjanssens is it possible to support this?

**Reply by johanjanssens** [reply]
@raramuridesign Just released Pages 0.21. The Extension API is now considered stable and i also released the Joomla extension. You can find more info here: https://github.com/joomlatools/joomlatools-pages/discussions/738

**Comment by johanjanssens** [comment]
Hello Matthew, sure, Pages has a complete extension API that allows you to integrate it quite easily with any Joomla extension.  

For Seblod it would depend a bit how exactly you want to make the integration work. Most Joomla extensions do not really offer their own API so the approach I used so far is to query the database directly. This gives the most flexibility and performance, keeps the code quite simple and allows to make use of all the power that Pages offers out of the box. 

What features exactly you need to get started?

---

For reference, the documentation for the Extension API can be found in the related PR's.

### Extension API

- Extension API: https://github.com/joomlatools/joomlatools-pages/pull/144
- Extension Packages: https://github.com/joomlatools/joomlatools-pages/pull/360
- Extension Router: https://github.com/joomlatools/joomlatools-pages/pull/680
- Extension Template Locator: https://github.com/joomlatools/joomlatools-pages/pull/369

Extension can also easily be bundled into phar's and be used without the need to unzip them, more info can be found in the following 2 PR's.

- Extension Phar Support: https://github.com/joomlatools/joomlatools-pages/pull/683
- Extension Management: https://github.com/joomlatools/joomlatools-pages/pull/689

### Extension Examples

I have been working on various extensions as examples how to use the API's. 

#### Joomla

- Code https://github.com/joomlatools/joomlatools-pages/tree/master/contrib/extensions/joomla
- Docs: https://github.com/joomlatools/joomlatools-pages/pull/363

This extension will be officially releases as part of the upcoming 0.21 release, the code has already been merged into master.

#### K2

 - Code: https://github.com/joomlatools/joomlatools-pages/tree/feature/364-K2/contrib/extensions/k2
 - Docs: https://github.com/joomlatools/joomlatools-pages/pull/365
 
 This extension is not fully updated to the latest code changes, and improvements, I have asked Fotis to help out but so far he hasn't been able to spend time on it.
 
 #### J2Store
 
 -  Code: https://github.com/joomlatools/joomlatools-pages/pull/377/files
 - https://github.com/joomlatools/joomlatools-pages/pull/377
 
 This extension has been deprecated due to a lack of interest. I asked the developers to help out but they haven't taken up on the request.
